core/src/apps/management/authenticate_device.py

```diff
@@ -11,10 +11,13 @@ async def authenticate_device(msg: AuthenticateDevice) -> AuthenticityProof:
     from trezor.crypto.hashlib import sha256
     from trezor.messages import AuthenticityProof
     from trezor.ui.layouts import confirm_action
-    from trezor.utils import BufferReader
+    from trezor.utils import BufferReader, bootloader_locked
 
     from apps.common.writers import write_compact_size
 
+    if not bootloader_locked():
+        raise wire.ProcessError("Bootloader unlocked.")
```

Comment:
```suggestion
        raise wire.ProcessError("Cannot authenticate when bootloader is unlocked.")
```
(otherwise it sounds like this action caused the bootloader to unlock)

Reply:
55238596fa265acc17e68d573bd4ac37d1cee290